packages/zosfiles/src/methods/create/Create.ts

```diff
@@ -142,6 +142,13 @@ export class Create {
         ImperativeExpect.toNotBeNullOrUndefined(dataSetName, ZosFilesMessages.missingDatasetName.message);
         ImperativeExpect.toNotBeNullOrUndefined(likeDataSetName, ZosFilesMessages.missingDatasetLikeName.message);
 
+        // Removes undefined properties
+        let tempOptions = !isNullOrUndefined(options) ? JSON.parse(JSON.stringify({ like: likeDataSetName, ...(options || {}) })) : {};
+        Create.dataSetValidateOptions(tempOptions);
+        const endpoint: string = ZosFilesConstants.RESOURCE + ZosFilesConstants.RES_DS_FILES + "/" + dataSetName;
+
+        const data = await ZosmfRestClient.postExpectString(session, endpoint, [], JSON.stringify(tempOptions));
```

Comment:
The `Create.datasetLike` API appears to be calling the z/OSMF endpoint twice. Please delete lines 153-155, and move lines 148-150 inside the `try` block. This should result in the desired behavior to call the endpoint only once with `tempOptions`.